waku/README.md

```diff
@@ -0,0 +1,301 @@
+- [Security Considerations](#security-considerations)
+- [Implementation](#implementation)
+- [Changelog](#changelog)
+    - [Differences between shh/6 waku/0](#differences-between-shh6-waku0-1)
+- [Copyright](#copyright)
+
+## Abstract
+
+This specification describes the format of Waku messages within the ÐΞVp2p Wire Protocol. This spec substitutes [EIP- 627](https://eips.ethereum.org/EIPS/eip-627). Waku is a fork of the original Whisper protocol that enables better scalability and offline messaging, at the cost of some metadata protection guarantees. It does this through (a) light node support (b) mailserver (c) waku mode and (d) basic rate limiting.
+
+## Motivation
+
+<!-- This should be the motivation of Waku rather than the motivation of this spec: aka to enable scalability etc. -->
+
+It is necessary to specify the standard for Waku messages in order to ensure forward compatibility of different Waku clients, backwards compatibility with Whisper clients, as well as to allow multiple implementations of Waku.
+
+## Specification
+
+### Use of RLPx transport protocol
+
+All Waku messages are sent as devp2p [RLPx transport protocol, version 5](https://github.com/ethereum/devp2p/blob/master/rlpx.md) packets. These packets SHOULD be RLP-encoded arrays of data containing two objects: packet code followed by another object (whose type depends on the packet code). 
+
+Waku nodes that do not support a particular packet code SHOULD ignore the packet without generating any error.
+
+Waku is a RLPx subprotocol called `waku` with version `0`. The version number corresponds to the major version in the header spec.
+
+### Packet Codes
+
+<!-- Packet code not clear -->
+
+The message codes reserved for Waku protocol: 0 - 127.
+
+<!-- Is this a MUST or rather a SHOULD -->
+
+Messages with unknown codes MUST be ignored, for forward compatibility of future versions.
+
+The Waku sub-protocol SHOULD support the following packet codes:
+
+| Name                       | Int Value |
+|----------------------------|-----------|
+| Status                     |     0     |
+| Messages                   |     1     |
+| PoW Requirement            |     2     |
+| Bloom Filter               |     3     |
+
+
+The following message codes are optional, but they are reserved for specific purpose.
+
+| Name                       | Int Value |
+|----------------------------|-----------|
+| P2P Request                |    126    |
+| P2P Message                |    127    |
+
+### Packet Format and Usage
+
+**Status** [`0`]
+
+<!-- @todo maybe make a table or something -->
+
+This packet contains two objects: integer message code (0x00) followed by a list of values: version, PoW requirement, and bloom filter, in this order. The bloom filter paramenter is optional; if it is missing or nil, the node is considered to be full node (i.e. accepts all messages). The format of PoW and bloom filter please see below (message codes 2 and 3).
```

Comment:
I noticed `integer` was removed at version. For PoW requirement and bloom filter this is explained lower. But I think that for version it should also be mentioned somewhere, even if it sounds trivial.

When we add light node boolean, also here it should be clarified which type. 

@adam perhaps you can give more details about the confirmations received part? Both implementation nor motivation why it is needed are clear to me.

Reply:
@kdeme readded